src/utilities/convex_hull.zig

```diff
@@ -291,7 +383,109 @@ fn rotateToXAxis(comptime Point: type, point: Point, angle: f64) ContinousPoint
+
+    // Iterate until a valid tangent is found
+    while (true) {
+        // Check the turn direction at hull_two
+        const hull_two_turn = computeTurn(
+            hull_one.items[hull_one_idx], // Current point in hull_one
+            hull_two.items[hull_two_idx], // Current point in hull_two
+            hull_two.items[(hull_two_idx + 1) % hull_two.items.len], // Next point in hull_two
+        );
+
+        // Adjust hull_two_idx based on the turn and hull type
+        switch (hull_type) {
+            .upperHull => if (hull_two_turn == Turn.left) {
+                // If the turn is left, the current tangent dips into hull_two
+                // Move to the next point in hull_two (clockwise adjustment)
+                hull_two_idx = (hull_two_idx + 1) % hull_two.items.len;
+                continue; // Re-evaluate the tangent
+            },
+            .lowerHull => if (hull_two_turn == Turn.right) {
+                // If the turn is right, the current tangent dips into hull_two
+                // Move to the next point in hull_two (clockwise adjustment)
+                hull_two_idx = (hull_two_idx + 1) % hull_two.items.len;
+                continue; // Re-evaluate the tangent
+            },
+        }
+
+        // Check the turn direction at hull_one
+        const hull_one_turn = computeTurn(
+            hull_one.items[(hull_one_idx - 1 + hull_one.items.len) % hull_one.items.len], // Previous point in hull_one
+            hull_one.items[hull_one_idx], // Current point in hull_one
+            hull_two.items[hull_two_idx], // Current point in hull_two
+        );
+
+        // Adjust hull_one_idx based on the turn and hull type
+        switch (hull_type) {
+            .upperHull => if (hull_one_turn == Turn.left) {
+                // If the turn is left, the current tangent dips into hull_one
+                // Move to the previous point in hull_one (counterclockwise adjustment)
+                hull_one_idx = (hull_one_idx - 1 + hull_one.items.len) % hull_one.items.len;
+                _ = hull_one.pop(); // Remove the point that violates convexity
+                continue; // Re-evaluate the tangent
+            },
+            .lowerHull => if (hull_one_turn == Turn.right) {
+                // If the turn is right, the current tangent dips into hull_one
+                // Move to the previous point in hull_one (counterclockwise adjustment)
+                hull_one_idx = (hull_one_idx - 1 + hull_one.items.len) % hull_one.items.len;
+                _ = hull_one.pop(); // Remove the point that violates convexity
+                continue; // Re-evaluate the tangent
+            },
+        }
+
+        // If neither index needs adjustment, we have found a valid tangent
+        break;
+    }
+
+    // Return the indices of the tangent points
+    return .{ .hull_one_idx = hull_one_idx, .hull_two_idx = hull_two_idx };
+}
+
+fn testConvexHullProperty(convex_hull: *ConvexHull) !void {
```

Comment:
Maybe move after the last test that use it so it is clear that it is not a function used in the implementation of the `ConvexHull`.

Reply:
fixed